float32 slope_left
float32 intercept_left
float32 slope_right
float32 intercept_right